### PR TITLE
feat: add an ability to view groups gf-78

### PR DESCRIPTION
**Pull request by s1rserg**
* Added groups table on Access Management page that displays Name, Permissions, and Created At columns.
* Added backend route for getting all groups.

image

### CODE REVIEW

File: apps/frontend/src/pages/access-management/access-management.tsx [hunk 1]
```diff
@@ -4,29 +4,39 @@ import {
...
+	const userColumns = getUserColumns();
+	const userData: UserRow[] = getUserRows(users);
+
+	const groupColumns = getGroupColumns();
+	const groupData: GroupRow[] = getGroupRows(groups);
 
 	return (
 		<PageLayout>
 			<p className={styles["title"]}>Access Management</p>
 			<p className={styles["sub-title"]}>Users</p>
```
Comment (liza-veis) [inline apps/frontend/src/pages/access-management/access-management.tsx:36]: Haven't noticed this previously, let's use appropriate tags and wrap each section with `section`
```suggestion
			<h1 className={styles["title"]}>Access Management</p>
			<section>
			  <h2 className={styles["section-title"]}>Users</p>
			  //...table
			<section>
```

File: packages/shared/src/modules/groups/libs/types/group-get-all-item-response-dto.type.ts [hunk 2]
```diff
@@ -0,0 +1,9 @@
+type GroupGetAllItemResponseDto = {
+	createdAt: string;
+	id: number;
+	name: string;
+	permissions: { id: number; name?: string }[];
```
Comment (liza-veis) [inline packages/shared/src/modules/groups/libs/types/group-get-all-item-response-dto.type.ts:5]: Why name is optional 🤔 
```suggestion
	permissions: { id: number; name: string }[];
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: apps/frontend/src/pages/access-management/access-management.tsx [hunk 3]
```diff
@@ -4,29 +4,39 @@ import {
 	useAppSelector,
 	useEffect,
 } from "~/libs/hooks/hooks.js";
+import { actions as groupActions } from "~/modules/groups/groups.js";
 import { actions as userActions } from "~/modules/users/users.js";
 
+import { getGroupColumns } from "./libs/helpers/get-group-columns.helper.js";
+import { getGroupRows } from "./libs/helpers/get-group-rows.helper.js";
 import { getUserColumns } from "./libs/helpers/get-user-columns.helper.js";
 import { getUserRows } from "./libs/helpers/helpers.js";
```
Comment (AnutkaGn) [inline apps/frontend/src/pages/access-management/access-management.tsx:13]: Import all helpers into a file helpers.ts and import everything from one place
```suggestion
import { getUserRows, getUserColumns, getGroupColumns, getGroupRows } from "./libs/helpers/helpers.js";
```

**Review by what1s1ove — APPROVED** [review]
Looking good after Lisa's PR changes

**Review by yevhenii-hladkov — APPROVED** [review]

File: apps/backend/src/modules/groups/group.entity.ts [hunk 4]
```diff
@@ -1,49 +1,52 @@
...
+	type GroupCreateResponseDto,
+	type GroupGetAllItemResponseDto,
+} from "./libs/types/types.js";
 
 class GroupEntity implements Entity {
+	private createdAt: null | string;
 	private id: null | number;
 	private name!: string;
-	private permissions!: Pick<PermissionModel, "id">[];
+	private permissions!: Array<{ id: number; name?: string }>;
```
Comment (liza-veis) [inline apps/backend/src/modules/groups/group.entity.ts:13]: Why did you change it this way instead of `Pick<PermissionModel, "id" | "name">[]`?

File: apps/backend/src/modules/groups/group.entity.ts [hunk 5]
```diff
@@ -52,34 +55,49 @@ class GroupEntity implements Entity {
...
 		return new GroupEntity({
+			createdAt: null,
 			id: null,
 			name,
 			permissions,
 			users,
 		});
 	}
 
+	public toGetAllItemObject(): GroupGetAllItemResponseDto {
```
Comment (liza-veis) [inline apps/backend/src/modules/groups/group.entity.ts:70]: We need to modify toObject instead, in our case we will need to have the same entity fields for get all as for create, because after creation we are going to insert returned item to the store
```suggestion
```

**Review by liza-veis — CHANGES_REQUESTED** [review]

File: apps/backend/src/modules/groups/group.entity.ts (hunk 4 above)
Comment (s1rserg) [reply apps/backend/src/modules/groups/group.entity.ts]: I needed to somehow state that name can be optional, since we don't receive it in payload and I can't provide those names to `initializeNew` in `group.service`.  I am now using `Array<Partial<Pick<PermissionModel, "name">> & Pick<PermissionModel, "id">>`.  Can't really find more elegant solution for this.

**Review by liza-veis — APPROVED** [review]
